cmd/entrypoint/runner.go

```diff
@@ -34,8 +37,51 @@ func (rr *realRunner) Run(args ...string) error {
 	defer signal.Reset()
 
 	cmd := exec.Command(name, args...)
+
 	cmd.Stdout = os.Stdout
+	var stdoutFile *os.File
+	if rr.stdoutFile != "" {
+		var err error
+		if stdoutFile, err = os.Create(rr.stdoutFile); err != nil {
+			return err
+		}
+		defer stdoutFile.Close()
+		// We use os.Pipe in asyncWriter to copy stdout instead of cmd.StdoutPipe or providing an
+		// io.Writer directly because otherwise Go would wait for the underlying fd to be closed by the
+		// child process before returning from cmd.Wait even if the process is no longer running. This
+		// would cause a deadlock if the child spawns a long running descendant process before exiting.
+		stopCh := make(chan struct{}, 1)
+		defer close(stopCh)
```

Comment:
I think we shouldn't need to create 2 `stopCh` (for `stdout` and `stderr`), on is enough :upside_down_face:

Reply:
Oh you're right. Thanks!